src/Sdk/DTWebApi/WebApi/TimelineRecord.cs

```diff
@@ -1,78 +1,86 @@
-﻿using System;
+using System;
```

Comment:
is this file changed in the actions service?

Reply:
I ported these changes to the actions service in a separate PR.  There were already significant divergences between the two copies.  As I mentioned above, longer-term I'd love to migrate all the replicated code into a library that can be shared between client and server.

It's less of a refactor than it is a baselining.  It wasn't strictly necessary, but it makes `TimelineRecord` much easier to reason about and maintain moving forward.

There's no longer any ambiguity about whether members are instantiated (non-null) or not.

For example, `record.ErrorCount++` is guaranteed to work correctly now.

Previously, the only way to guarantee a non-null outcome was to do something like:

```
record.ErrorCount = (record.ErrorCount ?? 0) + 1;
```